### PR TITLE
Improve trino session context

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

This pr improves the trino session context:
1. always reuse the kyuubi session if session id exists, so we can restore the session context for next query
2. transform trino client information to kyuubi session, e.g. trino request source (trino-cli)

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala [hunk 1]
```diff
@@ -140,15 +142,17 @@ object TrinoContext {
...
-    trinoContext.catalog.foreach(
-      responseBuilder.header(TRINO_HEADERS.responseSetCatalog, _))
-    trinoContext.schema.foreach(
-      responseBuilder.header(TRINO_HEADERS.responseSetSchema, _))
+    // Note, We have injected kyuubi session id to session context so that the next query can find
+    // the previous session to restore the query context.
+    // It's hard to follow the Trino style that set all context to http headers.
+    // Because we do not know the context at server side. e.g. `set k=v`, `use database`.
+    // We also can not inject other session context into header before we supporting to map
+    // query result to session context.
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:150]: cc @iodone  
We can not inject session context using the current request header. e.g. the current query is `set k=v2` but he exists request header is `k=v`. So we only inject kyuubi session id to help next query find session context.

Comment (iodone) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:150]: Instead of returning all session information to the client with HTTP response header, set it to sessionConfig and use kyuubi session to save the session information in server side?

Comment (ulysses-you) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:150]: yea, I think it's not the trino way but more suitable for Kyuubi

Comment (iodone) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala:150]: Nice, elegant solutions

**Comment by ulysses-you** [comment]
thanks, merging to master/1.7
